src/main/duckdb-c.cpp

```diff
@@ -92,6 +92,17 @@ void WriteData(duckdb_result *out, ChunkCollection &source, idx_t col) {
 	}
 }
 
+static bool duckdb_statement_return_changes(StatementType type) {
+	switch (type) {
+	case StatementType::INSERT_STATEMENT:
+	case StatementType::UPDATE_STATEMENT:
+	case StatementType::DELETE_STATEMENT:
+		return true;
```

Comment:
Since this appears duplicated from `sqlite3_statement_returns_changes` perhaps this can be unified in `statement_type.hpp`?

As a side note - CREATE can also return the number of rows inserted (in the case of `CREATE TABLE x AS SELECT 42`).